pkg/fileutils/fileutils_test.go

```diff
@@ -694,3 +694,73 @@ func TestMatch(t *testing.T) {
 		}
 	}
 }
+
+type compileTestCase struct {
+	pattern               string
+	matchType             matchType
+	compiledRegexp        string
+	windowsCompiledRegexp string
+}
+
+var compileTests = []compileTestCase{
+	{"*", regexpMatch, `^[^/]*$`, `^[^\\]*$`},
+	{"file*", regexpMatch, `^file[^/]*$`, `^file[^\\]*$`},
+	{"*file", regexpMatch, `^[^/]*file$`, `^[^\\]*file$`},
```

Comment:
We could potentially add match types to cover these cases without regexps, if you think it's worth it.

Reply:
(They could also cover `*a/b` and `a/b*`)